docs/manuals/source/DeveloperGuide/developerFaq.rst

```diff
@@ -0,0 +1,119 @@
+.. _developer-guide-faq:
+
+Developer FAQ
+=============
+
+How to run a job that automatically fails?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+I want to run a job that automatically fails to simulate failing jobs.
+
+To do so, you can use the directive :config:option:`dir/job/ClientRunBeforeJob`.
+This directive launches a script on the client-side ( file daemon ) before running the job.
+set the script to be “/bin/false”
+
+.. code-block:: bareosconfig
+   :caption: Job with client run before job directive
+
+   Job {
+       Name = "backup-bareos-fd"
+       JobDefs = "DefaultJob"
+       Client = "bareos-fd"
+       Client Run Before Job = /bin/false
+   }
+
+or, if you want a job to work properly but fail at the end (e.g: simulate a full backup that did not go well), you can do the following
+
+.. code-block:: bareosconfig
+   :caption: Job with failing runscript at the end
+
+   Job {
+     Name = "backup"
+     JobDefs = DefaultJob
+     RunScript {
+       Command = "/bin/false"
+       failjobonerror = yes
+       RunsWhen = After
+     }
+    }
+
+
+
+I have changes that affect tapes and autochangers, how can I test them?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In order to test tapes, we use mhvtl to simulate the presence of tape and autochanger devices.
+
+Documentation is available at https://www.mhvtl.com/
+
+First you need to clone the mhvtl repository from https://github.com/markh794/mhvtl.git.
+Then you should compile and install the mhvtl kernel module
+
+.. code-block:: shell-session
+
+   $ cd mhvtl/kernel
+   $ make
+   $ sudo make install
+
+After that you have to install the mhvtl user tools
+
+.. code-block:: shell-session
+
+   $ cd mhvtl
+   $ make
+   $ sudo make install
+
+
+Now that mhvtl is installed, you can enable and start the mhvtl services
+
+.. code-block:: shell-session
+
+   $ sudo systemctl enable mhvtl-load-modules.service
+   $ sudo systemctl start mhvtl-load-modules.service
+   $ sudo systemctl enable mhvtl.target
+   $ sudo systemctl start mhvtl.target
```

Comment:
The syntax to enable and start a service can be shortened like this with the `--now` flag
```suggestion
   $ sudo systemctl enable --now mhvtl-load-modules.service
   $ sudo systemctl enable --now mhvtl.target
```